test/OpenTelemetry.Tests/Trace/BatchExportActivityProcessorTest.cs

```diff
@@ -0,0 +1,113 @@
+﻿// <copyright file="BatchExportActivityProcessorTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using OpenTelemetry.Tests.Shared;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Trace
+{
+    public class BatchExportActivityProcessorTest
+    {
+        [Fact]
+        public void CheckNullExporter()
+        {
+            Assert.Throws<ArgumentNullException>(() => new BatchExportActivityProcessor(null));
+        }
+
+        [Fact]
+        public void CheckConstructorWithInvalidValues()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxQueueSize: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxExportBatchSize: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxExportBatchSize: 2049));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), scheduledDelayMilliseconds: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), exporterTimeoutMilliseconds: 0));
+        }
+
+        [Fact]
+        public void CheckIfBatchIsExportingOnQueueLimit()
+        {
+            using var exporter = new TestActivityExporter();
+            using var processor = new BatchExportActivityProcessor(exporter, maxQueueSize: 1, maxExportBatchSize: 1);
+            processor.OnStart(new Activity("start"));
+            processor.OnEnd(new Activity("start"));
+
+            while (true)
+            {
+                if (exporter.Exported.Count == 0)
+                {
+                    Thread.Sleep(500);
+                }
```

Comment:
In the event of a regression this test may hang in an infinite loop. In the past, I've found tests like this hard to diagnose. Can we wait up to a maximum period of iterations maybe?

Reply:
of course!

Reply:
done!